### PR TITLE
[Security] Pin keras~=3.11 to resolve safe_mode vulnerability (ML-11143)

### PR DESCRIPTION
**Pull request by omermaim**
### 📝 Description
Pin kera~=3.11 in the test environment to resolve 6 Dependabot security alerts (ML-11143). Keras `Model.load_model` silently ignores `safe_mode=True` for `.h5` files in versions < 3.11.3.  Keras is not a direct MLRun dependency — it enters the test environment transitively via TensorFlow. The fix is a version pin + lock file regeneration

---

### 🛠️ Changes Made
- Added `keras~=3.11` pin to `dev-requirements.txt` (frameworks
  tests section)
- Regenerated `dockerfiles/test/locked-requirements.txt` via `make
  upgrade-mlrun-test-deps-lock`

---

### ✅ Checklist
- [ ] I updated the documentation (if applicable)
- [x] I have tested the changes in this PR
- [x] I confirmed whether my changes are covered by system tests
  - [ ] If yes, I ran all relevant system tests and ensured they passed before submitting this PR
  - [ ] I updated existing system tests and/or added new ones if needed to cover my changes
- [ ] If I introduced a deprecation:
  - [ ] I followed the [Deprecation Guidelines](./DEPRECATION.md)
  - [ ] I updated the relevant Jira ticket for documentation
- [ ] Please run [Smoke-tests](https://github.com/mlrun/mlrun/actions/workflows/system-tests-opensource.yml) workflow, providing it the PR number as input (upon success, the workflow run will add label "Smoke tests: Pass" to the RP)

---

### 🧪 Testing
  
  

---

### 🔗 References
- Ticket link: https://iguazio.atlassian.net/browse/ML-11143
- Design docs links:
- External links:

---

### 🚨 Breaking Changes?

- [ ] Yes (explain below)
- [x] No

---

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -35,6 +35,8 @@ scikit-learn~=1.5.2
 lightgbm~=4.3
 xgboost~=1.1
 cryptography~=44.0
+# pin keras minimum to resolve safe_mode security vulnerability (ML-11143)
+keras~=3.11.3
```
Comment (liranbg) [inline dev-requirements.txt:39]: Hi, @omermaim - I dont see `keras` being installed / consumed before so I think this is pretty needless

**Review by liranbg — CHANGES_REQUESTED** [review]

File: dev-requirements.txt (hunk 1 above)
Comment (guy1992l) [reply dev-requirements.txt:39]: There are tests for it. If it was removed, I don't remember or know the reason behind it. It should be tested like xgboost and lightgbm and scikit-learn are tested (if they got removed as well then lol). 

Anyway, as you know I would love to deprecate `mlrun.frameworks`, all I care is for the security vulnerability to be removed, so if its not in any of the requirements, then this sec vul should be ignored.
